Dynamic_Bar.lua

```diff
@@ -19,34 +17,28 @@ Type:UsesAttributes("value, maxValue, valueColor")
 Type:UsesAttributes("state")
 Type:UsesAttributes("texture")
 
-TMW:RegisterUpgrade(45013, {
-	icon = function(self, ics)
-		if ics.Type == "dynamicbar" then
-			ics.Alpha = 1
-			ics.UnAlpha = ics.ConditionAlpha or 0
-			ics.ConditionAlpha = 0
-		end
-	end,
+local STATE_SHOW = TMW.CONST.STATE.DEFAULT_SHOW
+
+Type:RegisterConfigPanel_XMLTemplate(165, "TellMeWhen_IconStates", {
+	[STATE_SHOW]     = { order = 1, text = "|cFF00FF00" .. L["DEFAULT"], },
```

Comment:
Added the config panel for icon states so you can configure condition opacity. Also added a default state so the default appearance of the icon can be configured.